avr-gcc can't find limits.h

**Describe the bug**
avr-gcc is unable to find and include the header `<limits.h>`. Based on my limited understanding, it seems that avr-gcc does not provide this header itself, and instead delegates to the main system installation of gcc to include this header

from (avr-gcc's limits.h line 192):
```
#ifdef _GCC_NEXT_LIMITS_H
#include_next <limits.h>		/* recurse down to the real one */
#endif
```

**To Reproduce**
Create shell.nix:
```
{ pkgs ? import <nixpkgs> {} }:

pkgs.mkShell {
  buildInputs = with pkgs.pkgsCross; [
    avr.buildPackages.binutils
    avr.buildPackages.gcc
  ];
}
```
Create file.c:
```
#include <limits.h>
#include <stdio.h>

int main() { printf("%d\n", UINT_MAX); }
```
Enter nix shell with: `$ nix-shell shell.nix`

attempt compilation with: `$ avr-gcc file.c`

output:
```
In file included from /nix/store/sxdn938zndxp3jbj6qk29v4dp1z4v5l0-gcc-debug-8.3.0-avr-stage-final/lib/gcc/avr/8.3.0/include-fixed/syslimits.h:7,
                 from /nix/store/sxdn938zndxp3jbj6qk29v4dp1z4v5l0-gcc-debug-8.3.0-avr-stage-final/lib/gcc/avr/8.3.0/include-fixed/limits.h:34,
                 from file.c:1:
/nix/store/sxdn938zndxp3jbj6qk29v4dp1z4v5l0-gcc-debug-8.3.0-avr-stage-final/lib/gcc/avr/8.3.0/include-fixed/limits.h:194:15: fatal error: limits.h: No such file or directory
 #include_next <limits.h>  /* recurse down to the real one */
               ^~~~~~~~~~
compilation terminated.
```

--- a/pkgs/development/compilers/gcc/8/default.nix
+++ b/pkgs/development/compilers/gcc/8/default.nix
@@ -186,7 +186,12 @@ stdenv.mkDerivation ({
             sed -i gcc/config/linux.h -e '1i#undef LOCAL_INCLUDE_DIR'
         ''
         )
-    else "");
+    else "")
+      + stdenv.lib.optionalString targetPlatform.isAvr ''
+	        makeFlagsArray+=(
+	           'LIMITS_H_TEST=false'
+	        )
+	      '';
 
   inherit noSysDirs staticCompiler crossStageStatic
     libcCross crossMingw;

--- a/pkgs/development/compilers/gcc/9/default.nix
+++ b/pkgs/development/compilers/gcc/9/default.nix
@@ -185,7 +185,12 @@ stdenv.mkDerivation ({
             sed -i gcc/config/linux.h -e '1i#undef LOCAL_INCLUDE_DIR'
         ''
         )
-    else "");
+    else "")
+      + stdenv.lib.optionalString targetPlatform.isAvr ''
+	        makeFlagsArray+=(
+	           'LIMITS_H_TEST=false'
+	        )
+	      '';
 
   inherit noSysDirs staticCompiler crossStageStatic
     libcCross crossMingw;

